src/app/modules/shared/components/supplier/supplier.component.html

```diff
@@ -1,10 +1,16 @@
 <div class="table-background border">
-  <app-create-supplier (createSupplier)="addItem($event)" *ngIf="isVisible"></app-create-supplier>
-  <app-delete-supplier [supplier]="supplier" (deleteSupplier)="deleteSupplier($event)" *ngIf="isVisible"></app-delete-supplier>
-  <app-edit-supplier [supplier]="supplier" (updateSupplier)="updateSupplier($event)" *ngIf="isVisible"></app-edit-supplier>
   <p></p>
   <div>
-    <table id="supplier-table" class="table table-bordered  table-hover" style="width:100%;"></table>
+    <mat-fsp-table
+      [columnDefinitions]="columnDefinitions"
+      [columnNames]="columnNames"
+      [dataSource]="dataSource"
+      [numberOfRows]="numberOfRows">
```

Comment:
try to bind isVisible property of these modals as param to the table and change it  when click action button